### PR TITLE
feat(core): introduce CString clear method

### PR DESCRIPTION
**Pull request by dkrutskikh**
### **PR Type**
Enhancement

___

### **Description**
- Add `clear()` method to `CString` class

- Expand test coverage for `FixedString` modifiers

- Restructure tests with comprehensive sections

- Add Unicode and special character testing

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["CString class"] --> B["Add clear() method"]
  C["FixedString tests"] --> D["Expand clear tests"]
  C --> E["Expand insert tests"]
  C --> F["Expand erase tests"]
  B --> G["Complete modifier API"]
  D --> H["Comprehensive coverage"]
  E --> H
  F --> H
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>c_string.hpp</strong><dd><code>Add clear method declaration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

include/core/c_string.hpp

<ul><li>Add <code>clear()</code> method declaration with documentation<br> <li> Method marked as <code>constexpr</code> and <code>noexcept</code></ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/109/files#diff-5e47c64549828a3af3326bf67149f85cf5e336f9492239d74625826ff58bd56a">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>c_string.inl</strong><dd><code>Implement clear method</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

include/core/c_string.inl

<ul><li>Implement <code>clear()</code> method by setting <code>_data</code> to <code>_emptyString</code><br> <li> Simple constexpr implementation</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/109/files#diff-83492e82f06fe5d99ac148878a31e3bcb9392081ca4179a9595f8f109e5f87fd">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>c_string_test.cpp</strong><dd><code>Add comprehensive clear tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/core/c_string_test.cpp

<ul><li>Add comprehensive test suite for <code>clear()</code> method<br> <li> Test various scenarios: empty strings, single chars, Unicode<br> <li> Test multiple clear operations and reassignment</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/109/files#diff-61b83cabe56b2b5eca3eb29e11f0be5dd90045ed311cb9f511fc681ccadae5df">+166/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>fix_string_test.cpp</strong><dd><code>Enhance FixedString modifier tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/core/fix_string_test.cpp

<ul><li>Restructure existing tests into detailed sections<br> <li> Expand <code>clear()</code>, <code>insert()</code>, and <code>erase()</code> test coverage<br> <li> Add Unicode, special character, and edge case testing<br> <li> Remove old comment and improve test organization</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/109/files#diff-cc1ff39c1921067e1cd9ffcf0bc8c8f3d7349f0178f1eb6ddeac099bc0a2d2ae">+603/-112</a></td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: test/core/c_string_test.cpp [hunk 1]
```diff
@@ -1424,3 +1424,169 @@ TEST_CASE("CString capacity", "[core][c_string]") {
...
+    CString maxString("1234567890");
+
+    REQUIRE_FALSE(maxString.empty());
+    REQUIRE(std::strcmp(maxString.c_str(), "1234567890") == 0);
+
+    maxString.clear();
+
+    REQUIRE(maxString.empty());
+    REQUIRE(std::strcmp(maxString.c_str(), "") == 0);
+  }
```
[the remark this answers is not in the document]
Comment (dkrutskikh) [reply test/core/c_string_test.cpp:1487]: fixed

File: test/core/c_string_test.cpp [hunk 2]
```diff
@@ -1424,3 +1424,169 @@ TEST_CASE("CString capacity", "[core][c_string]") {
...
+    REQUIRE_FALSE(mixedString.empty());
+
+    unicodeString.clear();
+    emojiString.clear();
+    mixedString.clear();
+
+    REQUIRE(unicodeString.empty());
+    REQUIRE(emojiString.empty());
+    REQUIRE(mixedString.empty());
+  }
```
[the remark this answers is not in the document]
Comment (dkrutskikh) [reply test/core/c_string_test.cpp:1548]: fixed

File: test/core/c_string_test.cpp [hunk 3]
```diff
@@ -1424,3 +1424,169 @@ TEST_CASE("CString capacity", "[core][c_string]") {
...
+    REQUIRE_FALSE(specialString.empty());
+
+    newlineString.clear();
+    tabString.clear();
+    specialString.clear();
+
+    REQUIRE(newlineString.empty());
+    REQUIRE(tabString.empty());
+    REQUIRE(specialString.empty());
+  }
```
[the remark this answers is not in the document]
Comment (dkrutskikh) [reply test/core/c_string_test.cpp:1530]: fixed
